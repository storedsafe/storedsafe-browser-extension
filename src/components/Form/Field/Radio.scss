@import '../../../variables';

.field-radio {
  $size: 20px;
  $border-size: 2px;
  $padding: 2px;
  $inner-size: $size - ($border-size * 2) - ($padding * 2);

  .field-container {
    display: flex;
    flex-direction: column;
  }

  label {
    display: flex;
    align-items: center;
  }

  input {
    height: 0;
    opacity: 0;
    position: absolute;
    width: 0;
  }

  .custom-radio {
    padding: $padding;
    position: relative;
    border-radius: 100%;
    border: $border-size solid $border;
    box-shadow: 0 0 3px 0 rgba(0, 0, 0, 0.1) inset;
    width: $size;
    height: $size;
    margin-right: 0.5em;
  }

  .custom-radio::after {
    display: none;
    position: absolute;
    border-radius: 100%;
    content: '';
    width: $inner-size;
    height: $inner-size;
    background-color: $accent;
  }

  input:checked ~ .custom-radio::after {
    display: block;
  }

  input:hover ~ .custom-radio,
  input:active ~ .custom-radio,
  input:focus ~ .custom-radio {
    box-shadow: 0 0 3px 0 rgba(0, 0, 0, 0.1) inset, 0 0 3px 0 rgba(0, 0, 0, 0.1);
  }

  input:hover ~ .custom-radio {
    border-color: $border-hover;
    background-color: $bg-hover;
  }

  input:focus ~ .custom-radio {
    border-color: $border-focus;
    background-color: $bg-focus;
  }

  input:active ~ .custom-radio {
    border-color: $border-active;
    background-color: $bg-active;
  }
}
