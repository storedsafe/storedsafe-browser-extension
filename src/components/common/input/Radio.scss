@import '../../../styles/variables.scss';

.radio {
  $size: 16px;
  $border-size: 1px;
  $padding: 2px;
  $inner-size: $size - ($border-size * 2) - ($padding * 2);

  input {
    height: 0;
    opacity: 0;
    position: absolute;
    width: 0;
  }

  .custom-radio {
    padding: $padding;
    position: relative;
    border-radius: 100%;
    border: $border-size solid $color-border;
    box-shadow: 0 0 3px 0 rgba(0, 0, 0, 0.1) inset;
    width: $size;
    height: $size;
    box-sizing: border-box;
  }

  .custom-radio::after {
    display: none;
    position: absolute;
    border-radius: 100%;
    content: '';
    width: $inner-size;
    height: $inner-size;
    background-color: $color-accent-0;
  }

  input:checked ~ .custom-radio::after {
    display: block;
  }

  input:hover ~ .custom-radio,
  input:active ~ .custom-radio,
  input:focus ~ .custom-radio {
    box-shadow: 0 0 3px 0 rgba(0, 0, 0, 0.1) inset, 0 0 3px 0 rgba(0, 0, 0, 0.1);
  }

  input:hover ~ .custom-radio {
    border-color: $color-border-hover;
    background-color: $color-bg-hover;
  }

  input:focus ~ .custom-radio {
    border-color: $color-border-focus;
    background-color: $color-bg-hover;
  }

  input:active ~ .custom-radio {
    border-color: $color-border-active;
    background-color: $color-bg-active;
  }
}
