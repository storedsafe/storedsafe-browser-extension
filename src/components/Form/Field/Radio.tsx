import * as React from 'react';
import { FieldProps, FieldAttributes } from './Field';
import './Radio.scss';

type RadioProps = Pick<FieldProps, 'name' | 'options' | 'value' | 'onChange' | 'onBlur' | 'labelDir' | 'label'>;
export const Radio: React.SFC<RadioProps | FieldAttributes> = ({
  name,
  value,
  label,
  labelDir,
  options,
  onChange,
  onBlur,
  ...attributes
}: RadioProps) => (
  <div className="field field-radio">
    <fieldset className="field-control">
      <legend className="field-label">{label}</legend>
      <div className="field-container">
        {options.map((option) => (
          <label key={option.value} htmlFor={`${name}-${option.value}`}>
            {labelDir === 'left' && option.title}
            <input
              id={`${name}-${option.value}`}
              name={name}
              type="radio"
              value={option.value}
              checked={value === option.value}
              onChange={({ target }: React.ChangeEvent<HTMLInputElement>): void => {
                onChange && onChange(target.value, name);
              }}
              onBlur={({ target }: React.FocusEvent<HTMLInputElement>): void => {
                onBlur && onBlur(target.value, name);
              }}
              {...attributes}
            />
            <div className="custom-radio" />
            {labelDir === 'right' && option.title}
          </label>
        ))}
      </div>
    </fieldset>
  </div>
);

Radio.defaultProps = {
  options: [],
  labelDir: 'right',
};
