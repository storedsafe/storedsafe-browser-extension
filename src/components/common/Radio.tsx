import * as React from 'react';
import './Radio.scss';

type RadioProps = React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>;

export const Radio: React.SFC<RadioProps> = (props: RadioProps) => (
  <React.Fragment>
    <div style={{ display: 'inline-block' }} className="radio">
      <input type="radio" {...props} />
      <div className="custom-radio" />
    </div>
  </React.Fragment>
);
