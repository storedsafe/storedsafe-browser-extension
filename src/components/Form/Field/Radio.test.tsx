import * as React from 'react';
import { Radio } from './Radio';
import { shallow, mount } from 'enzyme';

const onChange = jest.fn();
const onBlur = jest.fn();

beforeEach(() => {
  onChange.mockClear();
  onBlur.mockClear();
});

const options = [
  { title: 'Foo', value: 'foo' },
  { title: 'Bar', value: 'bar' },
];

test('<Radio />', () => {
  const wrapper = shallow(<Radio name="test" label="Test" />);
  expect(wrapper.find('input#test').length).toBe(0);
  expect(wrapper).toMatchSnapshot();
});

test('<Radio options=...x2 />', () => {
  const wrapper = shallow(<Radio name="test" label="Test" options={options} />);
  const label = wrapper.find('label');
  expect(label.length).toBe(2);
  expect(label.at(0).text()).toBe(options[0].title)
  expect(label.at(1).text()).toBe(options[1].title)
  const input = wrapper.find('input[type="radio"]');
  expect(input.length).toBe(2);
  expect(input.at(0).props().value).toBe('foo')
  expect(input.at(1).props().value).toBe('bar')
  expect(wrapper).toMatchSnapshot();
});

test('<Radio options=...x2 labelDir=left />', () => {
  const wrapper = shallow(<Radio name="test" label="Test" options={options} labelDir="left" />);
  const label = wrapper.find('label');
  expect(label.length).toBe(2);
  expect(label.at(0).text()).toBe(options[0].title)
  expect(label.at(1).text()).toBe(options[1].title)
  const input = wrapper.find('input[type="radio"]');
  expect(input.length).toBe(2);
  expect(input.at(0).props().value).toBe('foo')
  expect(input.at(1).props().value).toBe('bar')
  expect(wrapper).toMatchSnapshot();
});


test('<Radio />, change', () => {
  const wrapper = mount(<Radio name="test" label="Test" options={options} value="foo" onChange={onChange} />);
  wrapper.find('input#test-bar').simulate('change', { target: { name: 'test', value: 'bar' } });
  expect(onChange).toHaveBeenCalledWith('bar', 'test');
});

test('<Radio />, blur', () => {
  const wrapper = mount(<Radio name="test" label="Test" options={options} value="foo" onBlur={onBlur} />);
  wrapper.find('input#test-bar').simulate('blur', { target: { name: 'test', value: 'bar' } });
  expect(onBlur).toHaveBeenCalledWith('bar', 'test');
});
