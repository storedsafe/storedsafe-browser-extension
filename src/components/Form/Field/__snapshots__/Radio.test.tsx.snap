// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> 1`] = `ShallowWrapper {}`;

exports[`<Radio options=...x2 /> 1`] = `ShallowWrapper {}`;

exports[`<Radio options=...x2 labelDir=left /> 1`] = `ShallowWrapper {}`;
